From 6c881297f5ab4f0b6744c2e3dab50e1d7e8710f1 Mon Sep 17 00:00:00 2001
From: Paul Smith <psmith@gnu.org>
Date: Sun, 12 Jun 2011 16:22:04 +0000
Subject: [PATCH 2/2] Fix another error related to whitespace handling in
 archives. Newer version of VMS support strncasecmp() so update the config.h.

Conflicts:
	ChangeLog
	tests/ChangeLog

Upstream-Status: Backport

---
 ChangeLog                       | 10 +++++++++-
 read.c                          | 10 +++++-----
 tests/ChangeLog                 |  6 ++++++
 tests/scripts/features/archives |  5 +++++
 5 files changed, 30 insertions(+), 6 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 2fc5df8..318b001 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,4 +1,12 @@
-=======
+2011-06-12  Paul Smith  <psmith@gnu.org>
+
+	* read.c (parse_file_seq): Move the check for empty members out of
+	the loop so we can go to the next member properly.
+	Another fix for Savannah bug #30612.
+
+	* config.h-vms.template: Newer versions of VMS have strncasecmp()
+	Patch provided by: Hartmut Becker <becker.ismaning@freenet.de>
+
 2010-08-13  Paul Smith  <psmith@gnu.org>
 
 	* NEWS: Accidentally forgot to back out the sorted wildcard
 
diff --git a/read.c b/read.c
index 9dfd4ea..c726aa4 100644
--- a/read.c
+++ b/read.c
@@ -3046,16 +3046,16 @@ parse_file_seq (char **stringp, unsigned int size, int stopchar,
                       nlen -= (n + 1) - tp;
                       tp = n + 1;
 
-                      /* If we have just "lib(", part of something like
-                         "lib( a b)", go to the next item.  */
-                      if (! nlen)
-                        continue;
-
                       /* We can stop looking now.  */
                       break;
                     }
                 }
               while (*e != '\0');
+
+              /* If we have just "lib(", part of something like "lib( a b)",
+                 go to the next item.  */
+              if (! nlen)
+                continue;
             }
         }
 
diff --git a/tests/ChangeLog b/tests/ChangeLog
index 7119c4c..c124b9a 100644
--- a/tests/ChangeLog
+++ b/tests/ChangeLog
@@ -1,3 +1,9 @@
+2011-06-12  Paul Smith  <psmith@gnu.org>
+
+	* scripts/features/archives: Check archives with whitespace at the
+	beginning, end, and extra in the middle.
+	Another test for Savannah bug #30612.
+
 2010-08-13  Paul Smith  <psmith@gnu.org>
 
 	* scripts/features/archives: New regression tests for archive
diff --git a/tests/scripts/features/archives b/tests/scripts/features/archives
index 00aa1af..3fe46a0 100644
--- a/tests/scripts/features/archives
+++ b/tests/scripts/features/archives
@@ -36,6 +36,11 @@ utouch(-50, 'a2.o');
 run_make_test('all: libxx.a(a3.o *.o)', '',
               "ar rv libxx.a a3.o\na - a3.o\nar rv libxx.a a2.o\nr - a2.o\n");
 
+# Check whitespace handling
+utouch(-40, 'a2.o');
+run_make_test('all: libxx.a(  a3.o    *.o     )', '',
+              "ar rv libxx.a a2.o\nr - a2.o\n");
+
 rmfiles(qw(a1.o a2.o a3.o libxx.a));
 
 # This tells the test driver that the perl test script executed properly.
-- 
1.8.1.4

